Redesign image captions

kdkasad opened the pull request:
Changes the style of image captions. This creates a distinciton between the `title` and `caption` attributes. The former now shows up more prominently than the latter. Both are centered below the image, which resolves #246.

Since there are no pages in the example site with images, I've added one in my fork's `exampleSite` branch: kdkasad@e7e62d98d4356c57bc9878e7cda8d5048f743f38. Below are screenshots of the various images on that page.

### Screenshots
#### Edit: captions are no longer centered. See [this comment](https://github.com/adityatelange/hugo-PaperMod/pull/247#issuecomment-781897919) for updated screenshots.

Cover image with caption:
![cover image](https://user-images.githubusercontent.com/63574107/108267331-e4b62800-711f-11eb-933f-71e12ca021b3.png)

Content image with title, caption, and attribution:
![image with title, caption, attr](https://user-images.githubusercontent.com/63574107/108267398-fe576f80-711f-11eb-899e-d5d8fe5c2300.png)

Content image with title and caption:
![image with title and caption](https://user-images.githubusercontent.com/63574107/108267486-1f1fc500-7120-11eb-9a9b-05a8ca641249.png)

Content image with title:
![image with title](https://user-images.githubusercontent.com/63574107/108267542-31016800-7120-11eb-90b7-219d09a77c2f.png)

Content image with caption:
![image with caption](https://user-images.githubusercontent.com/63574107/108267601-42e30b00-7120-11eb-835b-5cbcdb14b97f.png)

adityatelange commented:
The issue I thought for customizing captions in #127 inside the post was Cover image is meant to be full sized and fits the width.

While in case of images inside a post, the images can be smaller than width of `main` hence centering the caption and title does not seem a nice idea to me.

Ex. 

![image](https://user-images.githubusercontent.com/21258296/108365676-7b1c4380-721d-11eb-9899-efbb5aff6ad0.png)

kdkasad commented:
> The issue I thought for customizing captions in #127 inside the post was Cover image is meant to be full sized and fits the width.
> 
> While in case of images inside a post, the images can be smaller than width of `main` hence centering the caption and title does not seem a nice idea to me.
> 
> Ex.
> 
> ![image](https://user-images.githubusercontent.com/21258296/108365676-7b1c4380-721d-11eb-9899-efbb5aff6ad0.png)

I didn't think of that. Should I make the captions left-aligned as they were before, or should I try to center them to the image, even if the image is smaller than the page?

Jarmos-san commented:
Centering `figure` makes the most sense here. That way, even if the image is smaller in width than the rest of the page, it won't look weird for the reader.

adityatelange commented:
> Centering `figure` makes the most sense here. That way, even if the image is smaller in width than the rest of the page, it won't look weird for the reader.

I am sorry but this is your preference, not everyone wants this

kdkasad commented:
> > Centering `figure` makes the most sense here. That way, even if the image is smaller in width than the rest of the page, it won't look weird for the reader.
> 
> I am sorry but this is your preference, not everyone wants this

I completely understand. I've rebased the code and removed the center-alignment. Captions are now aligned according to the language's direction, which fixes the problems with images that are smaller than the page width. Here are some example images:

#### Left-to-right language direction
![ltr](https://user-images.githubusercontent.com/63574107/108473533-00661f00-7243-11eb-8707-edad1e190ab3.png)

#### Right-to-left language direction
![rtl](https://user-images.githubusercontent.com/63574107/108473457-e593aa80-7242-11eb-9fb8-bab48a2f4332.png)

#### Right-to-left language direction with small image
![rtl with small image](https://user-images.githubusercontent.com/63574107/108473777-55099a00-7243-11eb-88fb-cfaaec04b1a2.png)

Jarmos-san commented:
Well, these image orientations works for me too! I wouldn't mind that. Hugo's default formatting for `figure` just looks bland for PaperMod is what I wanted to say. 

**Centering it wasn't specifically for my needs, but was rather a suggestion** I thought would look nice with the theme. Apologies if my suggestions sounded otherwise.

adityatelange commented on assets/css/post-entry.css:
```diff
@@ -104,3 +104,7 @@
     box-shadow: 0 1px 0 var(--primary)
 }
+
+.entry-cover p {
+	margin: 8px 0 16px;
+}
```
Cover seems fine to me. Can you please remove this

kdkasad replied:
I had added this to make the cover image captions have the same padding as  the other image captions. I'm fine with removing it, but are you sure we don't want this?

**With padding:**
![caption with padding](https://user-images.githubusercontent.com/63574107/108542072-1ce07680-7298-11eb-961c-e3b38a528f2f.png)

**Without padding:**
![caption without padding](https://user-images.githubusercontent.com/63574107/108542105-29fd6580-7298-11eb-8650-6b6d5f2a2e30.png)

adityatelange replied:
yes

kdkasad replied:
I removed the second commit. Should be good to go now.

adityatelange reviewed (commented):
Everything else look good :)
